native/CMakeLists.txt

```diff
@@ -581,17 +544,44 @@ elseif(LINUX)
         cocos/platform/linux/modules/Accelerometer.h
         cocos/platform/linux/modules/Battery.cpp
         cocos/platform/linux/modules/Battery.h
-        cocos/platform/linux/modules/Screen.cpp
-        cocos/platform/linux/modules/Screen.h
         cocos/platform/linux/modules/Network.cpp
         cocos/platform/linux/modules/Network.h
         cocos/platform/linux/modules/Vibrator.cpp
         cocos/platform/linux/modules/Vibrator.h
         cocos/platform/linux/modules/System.cpp
         cocos/platform/linux/modules/System.h
+    )
+endif()
+
+############ desktop os USE_SERVER_MODE support
```

Comment:
I think it is better to put special operation into different platforms macros, for example
```
if(WINDOWS)
...
if(USE_SERVER_MODE)
        cocos/platform/empty/modules/Screen.cpp
        cocos/platform/empty/modules/Screen.h
        cocos/platform/empty/modules/SystemWindow.cpp
        cocos/platform/empty/modules/SystemWindow.h
esle()
    ...
#endif
endif()
```

Though it will have more codes, but more clear. Or it is strange that, windows doesn't includes Screen, Window in it s macro.